### PR TITLE
fix: 0 is falsey 🤦‍♂️

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Fixes a bug where `hasNegativeRatio` would be false if `inputOutputRatio` was `0`.

Relates to a `lib` fix: https://github.com/shapeshift/lib/pull/1230, however it's not a dependancy and this PR can go in without it.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Very small.

## Testing

Couple with an incoming `lib` PR. TODO.

## Screenshots (if applicable)

N/A

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4177** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4177" target="_blank">Graphite</a>  👈
    * **PR #4159** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4159" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4177?utm_source=stack-comment).

File: src/components/Trade/Components/TradeQuotes/TradeQuote.tsx [hunk 1]
```diff
@@ -128,12 +128,16 @@ export const TradeQuoteLoaded: React.FC<TradeQuoteLoadedProps> = ({
...
-  const amountEntered = bnOrZero(amount).gt(0)
-  const negativeRatio = !!inputOutputRatio && amountEntered && inputOutputRatio <= 0
-  const hasAmountWithPositiveReceive = amountEntered && !negativeRatio
+  const isAmountEntered = bnOrZero(amount).gt(0)
+  const hasNegativeRatio =
+    inputOutputRatio !== undefined && isAmountEntered && inputOutputRatio <= 0
+  const hasAmountWithPositiveReceive =
+    isAmountEntered &&
+    !hasNegativeRatio &&
+    bnOrZero(totalReceiveAmountCryptoPrecision).isGreaterThan(0)
```
Comment (0xApotheosis) [inline src/components/Trade/Components/TradeQuotes/TradeQuote.tsx:137]: In theory, if the `inputOutputRatio` is positive, `totalReceiveAmountCryptoPrecision` will be too.

In practice, we don't currently use the quote values (rather we compute them ourselves and incur a precision loss), and so it doesn't hold for very small values.

File: src/components/Trade/Components/TradeQuotes/TradeQuote.tsx [hunk 2]
```diff
@@ -128,12 +128,16 @@ export const TradeQuoteLoaded: React.FC<TradeQuoteLoadedProps> = ({
     : undefined
 
   const protocol = swapperWithMetadata.swapper.name
-  const amountEntered = bnOrZero(amount).gt(0)
-  const negativeRatio = !!inputOutputRatio && amountEntered && inputOutputRatio <= 0
-  const hasAmountWithPositiveReceive = amountEntered && !negativeRatio
+  const isAmountEntered = bnOrZero(amount).gt(0)
+  const hasNegativeRatio =
+    inputOutputRatio !== undefined && isAmountEntered && inputOutputRatio <= 0
```
Comment (0xApotheosis) [inline src/components/Trade/Components/TradeQuotes/TradeQuote.tsx:133]: `inputOutputRatio !== undefined` is the meat of this PR.

**Review by gomesalexandre — COMMENTED** [review]
See [review](https://github.com/shapeshift/lib/pull/1230#pullrequestreview-1368556558) from https://github.com/shapeshift/lib/pull/1230, pre-stamp, will stamp this as soon as the lib PR gets in

**Review by gomesalexandre — APPROVED** [review]
Sanity retested this after bumping the swapper dep, confirmed I wasn't able to see negative amounts for CoW (and miner fees are properly displayed for ZRX/Thor)

**Pull request merged**